### PR TITLE
feat(kubectl rollout): support multiple resources for rollout status

### LINKED ISSUE #72216: kubectl rollout status should support multiple resources
**What happened**:
```
deployments/
     a.yml
     b.yml
```

`kubectl rollout status -R -f deployments` fails with `error: rollout status is only supported on individual resources and resource collections - 2 resources were found`.

**What you expected to happen**:

The [docs](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-em-status-em-) suggest that `-R` should work for multiple resources in a directory when running `rollout status`. 

`status` should've been applied to all resources under `deployments` following https://github.com/kubernetes/kubernetes/issues/20435. `status` was omitted in https://github.com/kubernetes/kubernetes/pull/25110 and I wasn't able to find a discussion regarding this omission in other issues.

**How to reproduce it (as minimally and precisely as possible)**:

`kubectl rollout status -R -f deployments`

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.10.7
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/kind bug

### PR DESCRIPTION
**Pull request by phiott**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR adds support for multiple resources in `kubectl rollout status` (including `-R`),e.g. 

```
minikube-example ../kubernetes/_output/bin/kubectl rollout status -f deployments                                                                                                 (minikube/default)
deployment "example1" successfully rolled out
deployment "example2" successfully rolled out
```

where before this would fail with 

```
error: rollout status is only supported on individual resources and resource
```

#### Which issue(s) this PR fixes:

Fixes #72216

#### Special notes for your reviewer:

This is my first k8s PR - should I add unit tests for `kubectl rollout status` in https://github.com/kubernetes/kubernetes/blob/master/test/cmd/generic-resources.sh#L359? Noticed we don't have any coverage there.

#### Does this PR introduce a user-facing change?

```release-note
kubectl: support multiple resources for kubectl rollout status
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by pjo256** [comment]
@eddiezane Any chance you'd be able to take a look?

File: staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status.go [hunk 1]
```diff
@@ -168,73 +168,67 @@ func (o *RolloutStatusOptions) Run() error {
 		LabelSelectorParam(o.LabelSelector).
 		FilenameParam(o.EnforceNamespace, o.FilenameOptions).
 		ResourceTypeOrNameArgs(true, o.BuilderArgs...).
-		SingleResourceType().
+		ContinueOnError().
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status.go:171]: Could you please elaborate what will happen if one of resources get error like in here https://github.com/kubernetes/kubernetes/pull/108777/files#diff-484e3b9bfad2141dd92637f42b9d227b2d7e91d0932bb85ec7cf098c20770691R394-R396.

Do resources, whose are not getting any error, stay in waiting stage? or we exit immediately?

Comment (pjo256) [reply staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status.go:171]: We run against every visited resource as opposed to exiting immediately on the first error, e.g.

```
_output/bin/kubectl rollout status -f . -R --timeout=10s 
deployment "echoserver-1" successfully rolled out
Waiting for deployment "echoserver-2" rollout to finish: 0 of 1 updated replicas are available...
deployment "echoserver-3" successfully rolled out
error: timed out waiting for the condition
```

While looking into other `rollout` commands, this looks consistent but I noticed these commands (`pause, resume, history`) have a `Flatten()` call prior to `Do()`. This omission doesn't seem to change the output 🤔

**Review by soltysh — APPROVED** [review]
/triage accepted
/priority backlog
/lgtm
/approve

**Pull request merged**